cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/ApacheKafkaTopicService.java

```diff
@@ -59,6 +72,37 @@ public synchronized Set<TopicConfig> loadTopics(
       throw new Exception("Cannot connect to cluster.");
     }
 
+    ClusterKeyIdentifier clusterKeyIdentifier =
+        new ClusterKeyIdentifier(environment, protocol, clusterIdentification);
+
+    boolean topicsLoadingStatus = false;
+    if (topicsLoadingStatusOfClusters.containsKey(clusterKeyIdentifier)) {
+      topicsLoadingStatus = topicsLoadingStatusOfClusters.get(clusterKeyIdentifier);
+    } else {
+      topicsLoadingStatusOfClusters.put(clusterKeyIdentifier, false);
+    }
+
+    if (topicsLoadingStatus) {
+      return LoadTopicsResponse.builder()
+          .loadingInProgress(true)
+          .topicConfigSet(new HashSet<>())
+          .build();
+    } else {
+      topicsLoadingStatusOfClusters.put(clusterKeyIdentifier, true);
+    }
+
+    if (resetCache || !cachedTopics.containsKey(clusterKeyIdentifier)) {
+      loadTopicsForCache(client, topics, clusterKeyIdentifier);
```

Comment:
Just wondering would it make sense to have this wrapped in one if statement?

if (resetCache || !cachedTopics.containsKey(clusterKeyIdentifier)) {
topicsLoadingStatusOfClusters.put(clusterKeyIdentifier, true);
      loadTopicsForCache(client, topics, clusterKeyIdentifier);
topicsLoadingStatusOfClusters.put(clusterKeyIdentifier, false);
}

Reply:
Have couple of other conditions setting to false above line 85. I think it wouldn't work.